ravenframework/SupervisedLearning/DMD/BOPDMD.py

```diff
@@ -0,0 +1,192 @@
+                                                   \item $0.0 < trial\_size < 1.0$, $int(trial\_size * m)$ many observations will be used per trial,
+                                                       where $m$ denotes the total number of data points observed.
+                                                 \end{itemize}
+                                                 """, default=0.6))
+    specs.addSub(InputData.parameterInputFactory("eig_sort", contentType=InputTypes.makeEnumType("eig_sort", "SortedType",
+                                                                                                        ["real", "abs", "imag", "auto"]),
+                                                 descr=r"""Method used to sort eigenvalues (and modes accordingly) when performing BOP-DMD. Available options are:
+                                                 \begin{itemize}
+                                                   \item \textit{real}, eigenvalues will be sorted by real part and then by imaginary part to break ties.
+                                                   \item  \textit{imag}, eigenvalues will be sorted by imaginary part and then by real part to break ties.
+                                                   \item  \textit{abs}, eigenvalues will be sorted by magnitude.
+                                                   \item  \textit{auto}, one of the previously-mentioned sorting methods is chosen depending on eigenvalue variance.
+                                                 \end{itemize}
+                                                 """, default="auto"))
+    specs.addSub(InputData.parameterInputFactory("eig_constraints", contentType=InputTypes.makeEnumType("eig_constraints", "EigenConstraintType",
+                                                                                                        ["stable", "imag", "conjugate_pairs", "None"]),
+                                                 descr=r"""Set containing desired DMD operator eigenvalue constraints.. Available options are:
+                                                 \begin{itemize}
+                                                   \item \textit{stable}, constrains eigenvalues to the left half of the complex plane.
+                                                   \item  \textit{imag}, constrains eigenvalues to the imaginary axis.
+                                                   \item  \textit{conjugate_pairs}, enforces that eigenvalues are always present with their complex conjugate.
+                                                 \end{itemize}
+                                                 """, default=None))
+    return specs
+
+  def _handleInput(self, paramInput):
+    """
+      Function to handle the common parts of the distribution parameter input.
+      @ In, paramInput, ParameterInput, the already parsed input.
+      @ Out, None
+    """
+    import pydmd
```

Comment:
this is done here so we can keep the "lazy import" on the top of the file (to avoid to import the library if an object depending on it is not used in the input file)....it should accelerate the initialization of the framework (for cases where the dmd is not used)